src/optimizer/join_order/plan_enumerator.cpp

```diff
@@ -0,0 +1,591 @@
+#include "duckdb/optimizer/join_order/plan_enumerator.hpp"
+#include "duckdb/optimizer/join_order/query_graph_manager.hpp"
+
+namespace duckdb {
+
+class JoinNode;
```

Comment:
Usually, we have some forward declarations in the header file, so we don't need to include these files in the header. But since you are using `JoinNode` in this implementation file, I don't this this forward declaration is necessary.